### PR TITLE
Scrs 3002 officer date of birth

### PR DESCRIPTION
**Pull request by darrell-plant**

### CODE REVIEW

File: app/controllers/vatLodgingOfficer/OfficerNinoController.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    } yield Ok(views.html.pages.vatLodgingOfficer.officer_nino(res))
+
+  })
+
+  def submit: Action[AnyContent] = authorised.async(implicit user => implicit request => {
+    form.bindFromRequest().fold(
+      formWithErrors => BadRequest(views.html.pages.vatLodgingOfficer.officer_nino(formWithErrors)).pure,
+      data => {
+        viewModel[VoluntaryRegistration]
+          .map(_ == VoluntaryRegistration.yes).getOrElse(true).ifM(
```
Comment (PeterPerhac) [inline app/controllers/vatLodgingOfficer/OfficerNinoController.scala]: Here the  `s4l.saveForm[OfficerNinoView](data)` is repeated for both true/false cases

File: app/models/api/ScrsAddress.scala [hunk 2]
```diff
@@ -45,7 +45,7 @@ case class ScrsAddress(
     case _ => false
   }
 
-  override def hashCode: Int = asLabel.##
+  override def hashCode: Int = 1 // TODO temporary fix to ensure List.distinct works
```
Comment (PeterPerhac) [inline app/models/api/ScrsAddress.scala]: what do we do about this one?

File: app/models/view/vatLodgingOfficer/OfficerNinoView.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+case class OfficerNinoView(nino: String)
+
+object OfficerNinoView {
+
+  implicit val format = Json.format[OfficerNinoView]
+
+  // return a view model from a VatScheme instance
+  implicit val modelTransformer = ApiModelTransformer[OfficerNinoView] { vs: VatScheme =>
+    vs.lodgingOfficer.map(_.nino).collect {
+      case n: String => OfficerNinoView(n)
```
Comment (PeterPerhac) [inline app/models/view/vatLodgingOfficer/OfficerNinoView.scala]: we're collecting when n is String? could it be anything else? could we just do a `map` here?

Comment (darrell-plant) [inline app/models/view/vatLodgingOfficer/OfficerNinoView.scala]: yeaj, map is enough (good spot) 
vs.lodgingOfficer.map(_.nino).map(OfficerNinoView(_))

File: app/controllers/vatLodgingOfficer/OfficerNinoController.scala (hunk 1 above)
Comment (darrell-plant) [inline app/controllers/vatLodgingOfficer/OfficerNinoController.scala]: yep, fixing that...

File: app/models/api/ScrsAddress.scala (hunk 2 above)
Comment (darrell-plant) [inline app/models/api/ScrsAddress.scala]: it's a temporary fix - we should discuss

**Pull request merged**